osh/hub/osh_xmlrpc/worker.py

```diff
@@ -101,19 +112,46 @@ def finish_task(request, task_id):
     exclude_dirs = AppSettings.settings_get_results_tb_exclude_dirs()
     td = TaskResultsProcessor(task, base_task, exclude_dirs)
     td.unpack_results()
+
     if base_task:
         try:
-            return td.generate_diffs()
+            td.generate_diffs()
         except RuntimeError as ex:
             logger.error("Can't diff tasks %s %s: %s", base_task, task, ex)
             if not task.is_failed():
                 task.fail_task()
 
+    if settings.ENABLE_FEDORA_MESSAGING:
+        import django.urls
+        import os.path
+
+        scan_results_js_path = django.urls.reverse("task/log", args = [task_id,
+                                                                      os.path.join(task.args['result_filename'],
+                                                                                   "scan-results.js")])
+        scan_results_js_path += "?format=raw"
+        scan_results_js_url = request.build_absolute_uri(scan_results_js_path)
+
+        body = {'scan-results.js': scan_results_js_url}
+        if base_task:
+            added_js_path = django.urls.reverse("task/log", args = [task_id, "added.js"])
+            added_js_path += "?format=raw"
+            added_js_url = request.build_absolute_uri(added_js_path)
+            body.update({'added.js': added_js_url})
+
+            fixed_js_path = django.urls.reverse("task/log", args = [task_id, "fixed.js"])
+            fixed_js_path += "?format=raw"
+            fixed_js_url = request.build_absolute_uri(fixed_js_path)
+            body.update({'fixed.js': fixed_js_url})
+
+        publish_fedora_message(request, u'task.finish', task_id, body)
+
 
 @validate_worker
 def open_task(request, task_id):
     response = kobo_open_task(request, task_id)
 
+    publish_fedora_message(request, u'task.open', task_id)
```

Comment:
This would send messages for main task as well as the subtask. Shall we only publish messages for the main tasks?

Reply:
Yes.  I do not think that Packit or users themselves need to be notified when a subtask has changed state.